dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApi.java

```diff
@@ -0,0 +1,453 @@
+import org.jclouds.dimensiondata.cloudcontrol.options.PaginationOptions;
+import org.jclouds.http.filters.BasicAuthentication;
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.json.Json;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.Transform;
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.util.List;
+
+@RequestFilters({ BasicAuthentication.class, OrganisationIdFilter.class })
+@Consumes(MediaType.APPLICATION_JSON)
+@Path("/{jclouds.api-version}/network")
+public interface NetworkApi {
+
+   @Named("network:deployNetworkDomain")
+   @POST
+   @Path("/deployNetworkDomain")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   Response deployNetworkDomain(@PayloadParam("datacenterId") String datacenterId, @PayloadParam("name") String name,
+         @PayloadParam("description") String description, @PayloadParam("type") String type);
+
+   @Named("network:getNetworkDomain")
+   @GET
+   @Path("/networkDomain/{id}")
+   @Fallback(NullOnNotFoundOr404.class)
+   NetworkDomain getNetworkDomain(@PathParam("id") String networkDomainId);
+
+   @Named("network:listNetworkDomainsWithName")
+   @GET
+   @Path("/networkDomain")
+   @Transform(ParseNetworkDomains.ToPagedIterable.class)
+   @ResponseParser(ParseNetworkDomains.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<NetworkDomain> listNetworkDomainsWithDatacenterIdAndName(
+         @QueryParam("datacenterId") String datacenterId, @QueryParam("name") String name);
+
+   @Named("network:list")
+   @GET
+   @Path("/networkDomain")
+   @ResponseParser(ParseNetworkDomains.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<NetworkDomain> listNetworkDomains(PaginationOptions options);
+
+   @Named("network:list")
+   @GET
+   @Path("/networkDomain")
+   @Transform(ParseNetworkDomains.ToPagedIterable.class)
+   @ResponseParser(ParseNetworkDomains.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<NetworkDomain> listNetworkDomains();
+
+   @Named("network:deleteNetworkDomain")
+   @POST
+   @Path("/deleteNetworkDomain")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(NullOnNotFoundOr404.class)
+   Response deleteNetworkDomain(@PayloadParam("id") String networkDomainId);
```

Comment:
generally jclouds uses `@Fallback(VoidOnNotFoundOr404.class)` and uses `void deleteNetworkDomain(...)`
do you think `Response` is more useful here?